src/rdkafka_partition.c

```diff
@@ -4421,6 +4421,86 @@ int rd_kafka_topic_partition_list_regex_cnt(
 }
 
 
+/**
+ * @brief Removes reges from the list and returns the list without any regex in
+ * it.
+ */
+void rd_kafka_topic_partition_list_remove_regex(
+    rd_kafka_topic_partition_list_t **rktparlist) {
+        rd_kafka_topic_partition_list_t *non_regex_rktparlist;
+        int i;
+        int regex_cnt = rd_kafka_topic_partition_list_regex_cnt(*rktparlist);
+        int non_regex_cnt = (*rktparlist)->cnt - regex_cnt;
+
+        non_regex_rktparlist = rd_kafka_topic_partition_list_new(non_regex_cnt);
+
+        for (i = 0; i < (*rktparlist)->cnt; i++) {
+                const rd_kafka_topic_partition_t *rktpar =
+                    &((*rktparlist)->elems[i]);
+                if (*rktpar->topic != '^')
+                        rd_kafka_topic_partition_list_add_copy(
+                            non_regex_rktparlist, rktpar);
+        }
+
+        rd_kafka_topic_partition_list_destroy(*rktparlist);
+        *rktparlist = non_regex_rktparlist;
+}
+
+
+/**
+ * @brief Combine regex present in the list into a single regex.
+ */
+rd_kafkap_str_t *rd_kafka_topic_partition_list_combine_regex(
+    const rd_kafka_topic_partition_list_t *rktparlist) {
+        int i;
+        int combined_regex_len = 1; /* 1 for null-terminator */
+        int regex_cnt          = 0;
+        int j                  = 1;
+        char *combined_regex_str;
+        rd_kafkap_str_t *combined_regex_kstr;
+
+        // Count the number of characters needed for the combined regex string
+        for (i = 0; i < rktparlist->cnt; i++) {
+                const rd_kafka_topic_partition_t *rktpar =
+                    &(rktparlist->elems[i]);
+                if (*rktpar->topic == '^') {
+                        combined_regex_len += strlen(rktpar->topic);
+                        regex_cnt++;
+                }
+        }
+
+        if (regex_cnt == 0)
+                return NULL;
+
+        combined_regex_len += regex_cnt - 1; /* 1 for each '|' separator */
```

Comment:
We need to add `3 * (regex_cnt - 1)` chars here we have to combine with `)|(` as it's possible a regex is like `^a|b` and another one is `^c` and the combined one would be `(^a|b|^c)`. There shouldn't be a problem as `b` is matched in the first one as well, but I think it's safer (and more evident) if we write `(^a|b)|(^c)`